include/yaramod/types/token.h

```diff
@@ -0,0 +1,460 @@
+	HEX_WILDCARD_HIGH,
+	HEX_JUMP_LEFT_BRACKET, // '['
+	HEX_JUMP_RIGHT_BRACKET, // ']'
+	HEX_ALT_LEFT_BRACKET, // '('
+	HEX_ALT_RIGHT_BRACKET, // ')'
+	HEX_JUMP_FIXED,
+	HEX_START_BRACKET, // '{'
+	HEX_END_BRACKET, // '}'
+	NEW_LINE,
+	META, // 'meta'
+	MODIFIER,
+	LQUOTE,
+	RQUOTE,
+	RULE_END, // '}'
+	RULE_BEGIN, // '{'
+	RANGE,
+	DOT,
+	DOUBLE_DOT,
+	LT,
+	GT,
+	LE,
+	GE,
+	EQ,
+	NEQ,
+	SHIFT_LEFT,
+	SHIFT_RIGHT,
+	MINUS,
+	PLUS,
+	MULTIPLY,
+	DIVIDE,
+	MODULO,
+	BITWISE_XOR,
+	BITWISE_AND,
+	BITWISE_OR,
+	BITWISE_NOT,
+	LP,
+	RP,
+	LCB, // '{'
+	RCB, // '}'
+	ASSIGN,
+	COLON,
+	COMMA,
+	PRIVATE,
+	GLOBAL,
+	NONE,
+	RULE,
+	STRINGS,
+	CONDITION,
+	ASCII,
+	NOCASE,
+	WIDE,
+	FULLWORD,
+	XOR,
+	IMPORT_MODULE,
+	IMPORT_KEYWORD,
+	NOT,
+	AND,
+	OR,
+	ALL,
+	ANY,
+	OF,
+	THEM,
+	FOR,
+	ENTRYPOINT,
+	OP_AT,
+	OP_IN,
+	FILESIZE,
+	CONTAINS,
+	MATCHES,
+	SLASH,
+	STRING_LITERAL,
+	INTEGER,
+	DOUBLE,
+	STRING_ID,
+	STRING_ID_WILDCARD,
+	STRING_LENGTH,
+	STRING_OFFSET,
+	STRING_COUNT,
+	ID,
+	INTEGER_FUNCTION,
+	LSQB, // '['
+	RSQB, // ']'
+	DASH, // '-'
+	REGEXP_OR,
+	REGEXP_ITER,
+	REGEXP_PITER,
+	REGEXP_OPTIONAL,
+	REGEXP_START_SLASH,
+	REGEXP_END_SLASH,
+	REGEXP_CHAR,
+	REGEXP_RANGE,
+	REGEXP_TEXT,
+	REGEXP_CLASS_NEGATIVE,
+	REGEXP_MODIFIERS,
+	REGEXP_GREEDY,
+	UNARY_MINUS,
+	META_KEY,
+	META_VALUE,
+	STRING_KEY,
+	VALUE_SYMBOL,
+	FUNCTION_SYMBOL,
+	ARRAY_SYMBOL,
+	DICTIONARY_SYMBOL,
+	STRUCTURE_SYMBOL,
+	LP_ENUMERATION,
+	RP_ENUMERATION,
+	LP_WITH_SPACE_AFTER,
+	RP_WITH_SPACE_BEFORE,
+	LP_WITH_SPACES,
+	RP_WITH_SPACES,
+	BOOL_TRUE,
+	BOOL_FALSE,
+	ONELINE_COMMENT,
+	COMMENT,
+	INCLUDE_DIRECTIVE,
+	INCLUDE_PATH,
+	FUNCTION_CALL_LP,
+	FUNCTION_CALL_RP,
+	INVALID,
+};
+
+class Location
+{
+public:
+	Location() : Location(1, 0) {}
+	Location(std::size_t line, std::size_t column) : _begin(line, column), _end(line, column) {}
+
+	/// @name Modifiing methods
+	/// @{
+	void addLine(std::size_t count = 1)
+	{
+		std::swap(_begin, _end);
+		_end.first = _begin.first + count; // line
+		_end.second = 0; // column
+	}
+
+	void addColumn(std::size_t count)
+	{
+		_begin.first = _end.first;
+		_begin.second = _end.second;
+		_end.second += count;
+	}
+
+	void reset()
+	{
+		_begin = {1, 0};
+		_end = {1, 0};
+	}
+	/// @}
+
+	/// @name Getters
+	/// @{
+	std::pair<std::size_t, std::size_t> begin() const { return {_begin.first, _begin.second + 1}; }
+	const std::pair<std::size_t, std::size_t>& end() const { return _end; }
+	/// @}
+
+private:
+	std::pair<std::size_t, std::size_t> _begin; // (line, column)
+	std::pair<std::size_t, std::size_t> _end; // (line, column)
+};
+
+class Symbol;
+
+/**
+ * Class representing literal. Literal can be either
+ * string, integer or boolean. This class can only bear
+ * one literal type at the same time. Behavior is undefined
+ * when other type than actual type of the literal is requested.
+ *
+ * Caution: Integral literals are stored as string to preserve base
+ * and all preceding zeroes.
+ */
+class Literal
```

Comment:
Why can't `Literal` be in its own file?

Reply:
I moved it into separate file `literal.h` (and `literal.cpp`).